### ISSUE TITLE
[IDEA] Chapter 18: Basic DQN CartPole

### ISSUE THREAD

**Issue by IvanKuchin** [ageron/handson-ml2#519]
Hi Aurelien, 

Thank you for your book and amount of efforts you put down. This is the book #1 I've ever read. 

I've a suggestion about basic-DQN training in Chapter 18 - CartPole example. 

In the book-code you are using very small learning rate 1e-3 and number of trainings is 600-50=550. IMO 550 gradient applications is not enough for RL-model with that learning rate. Even though graph in the book looks awesome , I've not been able to reproduce it with same hyperparameters. As a possibility code might be specific to seedings. 

I'd propose 2 changes to book code: 
1) increase learning rate (5e-3 or even 1e-2, like in PolicyGrad). Seems that CartPole  is the "hello world" of RL and quite tolerate to high LR
2) On each episode run multiple Learning Steps (this is partially matches DeepMind '13 Fixed Q-Learning targets paper, by running learning at each step)
```
    if episode > 50:
        for _ in range(50):
            training_step(batch_size)
```

Side effects. having same model for learning and evaluation still kind of chasing your own tail, but it will converge/oscillate independently of seeding.  

On the other side I totally understand willingness to show how RL-model could be trained with small amount of compute, but it might bring a lot of confusion to audience. If you'll set proper expectations from the get go it could remove many questions like: "my model is not converging", "what seed I should use"....... 
Though if model is tabular it completely different game.

**Comment by ageron** [comment]
Hi Ivan,

Thanks for your very kind words and your interesting suggestion. I'm currently writing the 3rd edition of the book (at chapter 13), but as soon as I reach chapter 18, I'll give your suggestion a try, it sounds like a great idea. 🙏

**Comment by IvanKuchin** [comment]
Sum of undiscounted rewards per episode.
![image](https://user-images.githubusercontent.com/26530162/147760098-fc2eaa1f-631c-4b48-b223-93e29ea30c4e.png)
